app/src/main/java/bg/zahov/app/data/repository/SettingsRepositoryImpl.kt

```diff
@@ -6,25 +6,54 @@ import bg.zahov.app.data.interfaces.SettingsRepository
 import io.realm.kotlin.notifications.ObjectChange
 import kotlinx.coroutines.flow.Flow
 
+/**
+ * Repository implementation for managing user settings data.
+ *
+ * This singleton class interacts with the database layer via [RealmManager]
+ * to perform actions such as retrieving, updating, and resetting settings.
+ *
+ * @property realm Handles database operations related to settings through [RealmManager].
+ */
 class SettingsRepositoryImpl : SettingsRepository {
+
     companion object {
         @Volatile
```

Comment:
Why is this Volatile?

Reply:
@HauntedMilkshake ☝️

Reply:
I read it off a guide from here ->[baeldung singletons](https://www.baeldung.com/kotlin/singleton-classes)
To sum up:  "The @Volatile annotation is needed to ensure that the instance property is updated atomically."

Reply:
1. Again, not sure what the use case is but normally you shouldn't need to use that.